crates/fuel-core/src/service/genesis.rs

```diff
@@ -153,7 +153,7 @@ fn init_coin_state(
     db: &mut Database,
     state: &Option<StateConfig>,
 ) -> anyhow::Result<MerkleRoot> {
-    let mut coins_tree = binary::in_memory::MerkleTree::new();
+    let mut coins_tree = binary::root_calculator::MerkleRootCalculator::new();
```

Comment:
I think it is better to revert changes in this file to avoid conflicts with https://github.com/FuelLabs/fuel-core/pull/1413